### DISCUSSION TITLE
where to add custom parameters for Twitch as of V4?

### DISCUSSION THREAD

**Discussion by MatD1** [nextauthjs/next-auth#2758] [Help]
I need to add extra parameters into the provider so I am able to request my stream key from Twitch docs [here](https://dev.twitch.tv/docs/api/reference#get-stream-key).

I did go to the docs to attempt this however it sends me [here](https://github.com/nextauthjs/next-auth/blob/main/src/providers/twitch.js) which I don't think works in this version as when I tried to login it sends me to the error page http://localhost:3000/api/auth/signin?error=OAuthSignin and my console has an error error: `message: "Cannot read property 'url' of undefined",`

When I compare the code from [here](https://github.com/nextauthjs/next-auth/blob/main/src/providers/twitch.js) to the twitch provider in the nextauth node modules they are quite different.

How do I add extra parameters into the provider in v4?

**Comment by balazsorban44** [comment]
This sounds very familiar to https://github.com/nextauthjs/next-auth/issues/2735#issue-994233252 :confused: 

I am aware of the currently wrong link as well, but I'm not sure about the easiest fix: https://github.com/nextauthjs/docs/pull/29

If you look for the v4 config: https://github.com/nextauthjs/next-auth/blob/beta/src/providers/twitch.ts

Interestingly enough, when I use `next-auth/providers/twitch` as is with:
```
Twitch({
  clientId: process.env.TWITCH_ID,
  clientSecret: process.env.TWITCH_SECRET
})
```

It signs me in correctly, with no errors. :shrug:

Could you maybe give a reproduction to the issue?
